### ISSUE TITLE
Allow merge with match of string field to upper field

### ISSUE THREAD

**Issue by charlesknightsbridge** [EpiData-dk/analysis#88]
Request of JL

File a with a string field could be merged with upper from file b 

Read a; 
Merge v1 !fn:="b";  
// allow this if v1 is string in a and upper in b

**Comment by torstenchr** [comment]
@JLepidata , @charlesknightsbridge : Are we really sure this is a good idea?

This will be a breaking change to existing programming and may potentially break code that is already in use. If a string field is used as an ID it could be that data on purpose is differentiated by using uppercase and/or lowercase  text.

If we do want this implement this I would suggest we do this as an option. An options that relaxed the strict field type checking between the two datasets.

I do not recommend we implement this as the default implementation.

**Comment by charlesknightsbridge** [comment]
I have to agree with Torsten. For most applications, it is easy enough to create a new string variable to use for the merge.
The option, if we implement it, should be !ignorecase 
However, I don't know how well this would work in languages that don't use a roman alphabet. I suppose we would use the FPC function UnicodeSameStr(a,b).

**Comment by torstenchr** [comment]
Lazarus and/or FPC has a set of string function to handle UTF-8 uppercase/lowercase comparison and change. It would not be a very difficult option to implement, but it would be nice to have some tests we can add to our Validate repo such that we can ensure this functionality in the future as well.
